addons/binding/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/job/DailyJob.java

```diff
@@ -8,21 +8,139 @@
  */
 package org.openhab.binding.astro.internal.job;
 
+import static org.openhab.binding.astro.AstroBindingConstants.*;
+import static org.quartz.JobBuilder.newJob;
+import static org.quartz.TriggerBuilder.newTrigger;
+
+import java.util.Calendar;
+
+import org.apache.commons.lang.time.DateFormatUtils;
 import org.openhab.binding.astro.handler.AstroThingHandler;
 import org.openhab.binding.astro.internal.AstroHandlerFactory;
+import org.openhab.binding.astro.internal.model.Eclipse;
+import org.openhab.binding.astro.internal.model.Moon;
+import org.openhab.binding.astro.internal.model.MoonPhase;
+import org.openhab.binding.astro.internal.model.Planet;
+import org.openhab.binding.astro.internal.model.Range;
+import org.openhab.binding.astro.internal.model.Sun;
+import org.openhab.binding.astro.internal.model.SunEclipse;
+import org.openhab.binding.astro.internal.util.DateTimeUtils;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.Trigger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * Calculates and publishes the Sun data.
- * 
+ * Calculates and publishes the planet data and also schedules the events for the current day.
+ *
  * @author Gerhard Riegler - Initial contribution
  */
 public class DailyJob extends AbstractBaseJob {
+    private static final Logger logger = LoggerFactory.getLogger(DailyJob.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void executeJob(String thingUid, JobDataMap jobDataMap) {
```

Comment:
Could you extract the Moon- and Sun-specific parts of executeJob(...) into their own classes as it has been done with MoonThingHandler and SunThingHandler? Personally, I prefer polymorhism instead of big if/then/else switches.

Also, this would make things easier if we would ever capture an asteroid and put it into earth orbit as our secondary moon 😄